run/helm/snap/templates/daemonset.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: snap
+  namespace: {{ .Release.Namespace }}
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+spec:
+  template:
+    metadata:
+      name: snap
+      labels:
+        daemon: snapteld
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repo }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        securityContext:
+          privileged: {{ .Values.privileged }}
+        ports:
+        - containerPort: 8181
+        volumeMounts:
+        - name: autoload
+          mountPath: {{ .Values.config.autodiscover_path }}
+        - name: config
+          mountPath: /etc/snap
+        {{- range $key, $val := .Values.mounts }}
+        - name: {{ $key | replace "/" "" | replace "." "" }}
+          mountPath: {{ $val | quote }}
+        {{- end }}
+      volumes:
+      - name: autoload
+        hostPath:
```

Comment:
Okay, I am starting to understand the reasoning and I think this is going in the right direction, but I can still see some problems that someone managing a cluster would encounter.  The first is that Kubernetes is designed in a way that should ignore the underlying nodes.  The second advantage of `hostDir` that @mkuculyma points out, using custom tasks on each node, would require the Kubernetes user to tie specific kubernetes manifests to specific nodes.  

I don't really have a definitive solution to this problem, but just an idea.  I know this is outside of the scope of this specific PR, but what if we took advantage of annotations in Kubernetes and specified plugins and/or tasks in the Kubernetes manifests.  Tasks could be attached as config maps for easy management.